src/utils/SdkCoin.sol

```diff
@@ -0,0 +1,124 @@
+    {
+        // Note if we want to allow different cosmos decimals, we need to handle that here too
+        // Note that tokenAddress input validations are executed in the _getERC20TokenDecimals function
+        uint8 tokenDecimals = _getERC20TokenDecimals(tokenAddress);
+
+        if (tokenDecimals < 6) {
+            revert IISdkCoinErrors.UnsupportedTokenDecimals(tokenDecimals);
+        }
+        // Amount input validation
+        if (amount == 0) {
+            revert IISdkCoinErrors.ZeroAmountUint256(amount);
+        }
+        // Ensure the amount respects the token's decimals
+        // Handle the case where the input amount exceeds the token's precision
+        uint256 temp_convertedAmount;
+        uint256 factor;
+        uint256 remainder;
+        // Case ERC20 decimals are bigger than cosmos decimals
+        if (tokenDecimals > DEFAULT_COSMOS_DECIMALS) {
+            factor = 10 ** (tokenDecimals - DEFAULT_COSMOS_DECIMALS);
+            temp_convertedAmount = amount / factor;
+            remainder = amount % factor;
+        } else if (tokenDecimals == DEFAULT_COSMOS_DECIMALS) {
+            temp_convertedAmount = amount;
+            remainder = 0;
+        } else {
+            // revert as this is unreachable
+            revert IISdkCoinErrors.Unsupported();
+        }
+        return (SafeCast.toUint64(temp_convertedAmount), remainder);
+    }
+
+    // Convert Cosmos coin amount to ERC20 token amount
+    // Assuming that we support only ERC20.decimlas()>=6
+    /**
+     * @notice Convert the uint64 Cosmos coin amount into ERC20 token amount uint256
+     * @param tokenAddress The address of the token contract
+     * @param amount The amount to be converted
+     * @return convertedAmount The amount converted to uint256 supported by ERC20 tokens
+     */
+    function _convertSdkCoinAmountToERC20(address tokenAddress, uint64 amount) internal view returns (uint256) {
+        // Get the token decimals
+        // address input validation perfomed in the _getERC20TokenDecimals
+        uint8 tokenDecimals = _getERC20TokenDecimals(tokenAddress);
+
+        // Ensure the token has at least 6 decimals
+        if (tokenDecimals < 6) {
+            revert IISdkCoinErrors.UnsupportedTokenDecimals(tokenDecimals);
+        }
+        // Amount input validation
+        if (amount == 0) {
+            revert IISdkCoinErrors.ZeroAmountUint64(amount);
+        }
+        uint256 factor;
+        uint256 convertedAmount;
+        // Case ERC20 decimals are bigger than cosmos decimals
+        if (tokenDecimals > DEFAULT_COSMOS_DECIMALS) {
+            factor = 10 ** (tokenDecimals - DEFAULT_COSMOS_DECIMALS);
+            // uint256 = uint64 * uint256 that should be ok
+            convertedAmount = amount * factor;
+        } else if (tokenDecimals < DEFAULT_COSMOS_DECIMALS) {
+            // TODO if we decide to support this case. It will require handling the loss of precision
+            // in the go side
```

Comment:
Shouldn't this be reverting?

Also it can be reorganized as above, i.e.
```solidity
if (tokenDecimals > DEFAULT_COSMOS_DECIMALS) {
    // ...
else if (tokenDecimals == DEFAULT_COSMOS_DECIMALS) {
    // ...
} else {
    // revert
}
```